custom/enikshay/case_utils.py

```diff
@@ -170,6 +172,39 @@ def get_open_episode_case_from_person(domain, person_case_id):
     )
 
 
+def get_open_referral_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    open_referral_cases = [
+        case for case in reverse_indexed_cases
+        if not case.closed and case.type == CASE_TYPE_REFERRAL
+    ]
+    if not open_referral_cases:
+        return None
+    if len(open_referral_cases) == 1:
+        return open_referral_cases[0]
+    else:
+        raise ENikshayException(
+            "Expected none or one open referral case for person with id: {}".format(person_case_id)
+        )
+
+
+def get_latest_trail_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    trail_cases = [
+        case for case in reverse_indexed_cases
+        if case.type == CASE_TYPE_TRAIL
+    ]
+    trail_cases.sort(key=lambda c: c.opened_on)
```

Comment:
ah that's good context. didn't realize it was less than 100. i meant add a parameter to the actual sql function. see `get_forms_by_state.sql` for an example. you could potentially make a function that is get_ordered_cases.sql that accepts a column name to order by. then the CaseAccessor would pass in `opened_on`. then the couch and sql version would accept the same params. 

it's also probably fine to just hardcode opened_on in the sql function. i've just rewritten a few of those sql functions wishing i had made them more flexible. it doesn't seem super likely we'll ever need to order cases by anything else (maybe modified_on?)